implementations/micrometer-registry-wavefront/src/main/java/io/micrometer/wavefront/WavefrontHistogram.java

```diff
@@ -0,0 +1,178 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.wavefront;
+
+import com.wavefront.sdk.entities.histograms.WavefrontHistogramImpl;
+import io.micrometer.core.instrument.AbstractDistributionSummary;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
+import io.micrometer.core.instrument.distribution.HistogramSnapshot;
+import io.micrometer.core.lang.Nullable;
+
+import java.util.List;
+
+import static io.micrometer.wavefront.WavefrontConstants.WAVEFRONT_METRIC_TYPE_TAG_KEY;
+import static io.micrometer.wavefront.WavefrontConstants.isWavefrontMetricType;
+
+/**
+ * Wavefront's implementation of a DistributionSummary based off of WavefrontHistogramImpl.
+ *
+ * @author Han Zhang
+ */
+public class WavefrontHistogram extends AbstractDistributionSummary {
+    /**
+     * The tag value that is used to identify WavefrontHistograms.
+     */
+    private static final String WAVEFRONT_METRIC_TYPE_TAG_VALUE = "wavefrontHistogram";
+
+    private final WavefrontHistogramImpl delegate;
+
+    static Builder builder(String name) {
+        return new Builder(name);
+    }
+
+    /**
+     * @param id    The identifier for a metric.
+     * @return {@code true} if the id identifies a WavefrontHistogram, {@code false} otherwise.
+     */
+    static boolean isWavefrontHistogram(Id id) {
+        return isWavefrontMetricType(id, WAVEFRONT_METRIC_TYPE_TAG_VALUE);
+    }
+
+    WavefrontHistogram(Id id, Clock clock,
+                       DistributionStatisticConfig distributionStatisticConfig,
+                       double scale) {
+        super(id, clock, distributionStatisticConfig, scale, false);
+        delegate = new WavefrontHistogramImpl(clock::wallTime);
+    }
+
+    @Override
+    protected void recordNonNegative(double amount) {
+        delegate.update(amount);
+    }
+
+    @Override
+    public long count() {
+        return delegate.getCount();
+    }
+
+    @Override
+    public double totalAmount() {
+        return delegate.getSum();
+    }
+
+    @Override
+    public double mean() {
+        return delegate.getMean();
+    }
+
+    @Override
+    public double max() {
+        return delegate.getMax();
+    }
+
+    @Override
+    public HistogramSnapshot takeSnapshot() {
+        throw new UnsupportedOperationException(
+            "takeSnapshot() is not supported by WavefrontHistogram");
+    }
+
+    public List<WavefrontHistogramImpl.Distribution> flushDistributions() {
+        return delegate.flushDistributions();
+    }
+
+    /**
+     * Fluent builder for Wavefront histograms.
+     */
+    static class Builder {
```

Comment:
This builder should not be necessary. A user should never build an implementation-specific form of a meter.

Reply:
Added public `wavefrontHistogram(...)` methods to `WavefrontMeterRegistry`, so users will be using those instead of the builder.

Reply:
We cannot accept vendor-specific meter types. It breaks the portability of the instrumentation.